pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/ProperLogger.xml

```diff
@@ -87,6 +89,99 @@ bug 1626232, extra loggers with different names are not allowed by default (use
 public class Foo {
     private static final Log LOG = LogFactory.getLog(Foo.class);
     private static final Log LOG2 = LogFactory.getLog(Foo.class + ".foo");
+}
+ ]]>    </code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+Custom logger-class with invalid logger variable name
+        ]]></description>
+        <rule-property name="loggerClass">MyLog</rule-property>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    private static final MyLog INVALID_LOG_NAME = LogFactory.getLog(Foo.class);
+}
+ ]]>    </code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+Public logger
+        ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    public static final Log LOG = LogFactory.getLog(Foo.class);
+}
+ ]]>    </code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+Public logger when static final required
+        ]]></description>
+        <rule-property name="requireStaticFinalLogger">true</rule-property>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    public static final Log LOG = LogFactory.getLog(Foo.class);
+}
+ ]]>    </code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+Logger initialized as String constant from class name
+        ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    Log LOG = LogFactory.getLog(Foo.class.getName());
```

Comment:
Since the field is not private, the rule should should flag this... the test case should be `private static final Log LOG = LogFactory.getLog(Foo.class.getName());`.